brigade-worker/src/brigadier.ts

```diff
@@ -52,6 +53,11 @@ export class Job extends jobImpl.Job {
   run(): Promise<jobImpl.Result> {
     let jr = new JobRunner(this, currentEvent, currentProject);
     this._podName = jr.name;
+
+    if ("github" === currentEvent.provider) {
```

Comment:
I could imagine this block becoming a giant switch statement as more gateways are developed.

```javascript
switch(currentEvent.provider) {
  case "github":
    something();
  case "trello":
    somethingElse();
}
```

I wonder if there's a way that this logic could live within the gateway instead of within the worker? I kind of like the brigade-worker not needing to know about individual gateways. That would also allow authors of custom gateways to add this information to their respective payloads.

Reply:
Good point!

I agree that we would be better avoiding tight-coupling between worker<->gateway.

What I'm not clear about though is that how we could implement it cleanly.
What we could extend here would be only `currentEvent` and `currentProject`.
AFAIK, the latter former is passed as-is from gateways and the latter is passed as-is from the project config.
So - there would be no straight-forward way to provide envs to be used by workers from the gateway-side?

Reply:
Update: In my understanding, brigade works like this:

1. The GitHub gateway receives a webhook and [creates a corresponding `brigade.Build`](https://github.com/Azure/brigade/blob/f6c8b7aa3166d390852f0d34638424ca90e8ed6b/pkg/webhook/github.go#L213-L222)
2. BrigadeController receives the `brigade.Build` and [creates a worker to run it](https://github.com/Azure/brigade/blob/2d6281b12ed97d70ddb675dc79a5e70dc8722583/brigade-controller/cmd/brigade-controller/controller/handler.go#L46).

The step 2. is especially relevant. For example, we could add `Env` to the `brigade.Build` to be populated with [these](https://github.com/Azure/brigade/pull/259/files#diff-ca57cc779014810cd541e48951772a56R10) in the gateway, which is then passed to a worker container's `env` as-is. Do you think it is feasible?

Reply:
@technosophos Would you be ok with adding `Env` to `brigade.Build` as above?

Reply:
This isn't a blocker right now from my perspective, just something to consider.

Reply:
@dunglas Thanks for the sharing the context 👍 Yes, this seems like no blocker for me, too.

Perhaps once we get some time and/or another use-cases requiring `brigade.Build` to have  custom Env vars populated in a gateway, we can submit an another PR to do so?